data-rights-protocol.md

```diff
@@ -310,43 +367,48 @@ Note that these error states only represent *request errors*; workflow errors SH
 
-#### 3.07.1 JWT Key Semantics and Management
+We believe that providing these signed messages will ensure message integrity and prevent re-use or re-appropriation of requests: A data rights request represents a single action of one user acting against one covered business one time.
 
-In this protocol version we want to signal a move to *asymmetric key management* with unduly burdening implementers. Rather than relying on pair-wise symmetric keys multiplicatively exchanged between Authorized Agents and Covered Businesses, Authorized Agents SHALL generate 256-bit RSA keys and present their public portion in [RFC 7517](https://www.rfc-editor.org/rfc/rfc7517) JWK files.
+#### 3.07.1 Ed25519 Key Semantics, Request Signing, and Management
 
-Authorized Agents SHALL use JWTs with `alg` set to `RS256` RSASSA-PKCS1-v1_5 using SHA-256. RSA symmetric keys are well supported, easy to generate, and easy grasp conceptually. Future iterations of the protocol will use `libsodium`'s built in key generation utilities to provide cryptographic agility.
+The Data Rights Protocol authors **strongly** recommend the use of a [libsodium](https://doc.libsodium.org/)-based Ed25519 implementation. There is a [wide selection](https://doc.libsodium.org/bindings_for_other_languages) of language bindings for `libsodium` and in general is considered to be a high-quality, trustworthy API. Requests SHALL be signed by prepending the JSON document with the signature and sending both as a binary `application/octet-stream` request.
 
-Keys can be generated in a number of fashions:
-- [C implementation of JOSE](https://github.com/latchset/jose) can generate JWKs: `jose jwk gen -i '{"alg":"RS256"}' -o rsa.jwk && jose jwk pub -i rsa.jwk -o rsa.pub.jwk`
-- [MITRE's json-web-key-generator](https://github.com/mitreid-connect/json-web-key-generator) will generate the key from scratch
-- [pem2jwk](https://github.com/mt-inside/pem2jwks) will accept an existing PEM-encoded key and convert it to JWK
-- [jwkcreator](https://russelldavies.github.io/jwk-creator/) web site MAY be used to convert PEM keys to JWKs which will not be used to exchange 3rd party identity and rights
-- [mkjwk.org](https://mkjwk.org/) web site MAY be used to generate JWKs which will not be used to exchange 3rd party identity and rights.
+We provide in OSIRAA an example [generating keys and signing
+requests](https://github.com/consumer-reports-digital-lab/osiraa/blob/main/drp_aa_mvp/data_rights_request/pynacl_validator_submit.py)
+using the PyNaCl implementation, as well as a [Django request
+handler](https://github.com/consumer-reports-digital-lab/osiraa/blob/main/drp_aa_mvp/data_rights_request/pynacl_validator.py)
+which does cryptographic verification of the requests (without the full semantic validation chain presented above).
```

Comment:
I feel a fair bit itchy about giving people example security-critical code which isn't fully correct and complete, and well-commented. I know it's a fair bit of extra work, but I'd like to be able to give people samples with as few gotchas and caveats as possible.